src/main/java/io/atoti/spark/condition/AndCondition.java

```diff
@@ -14,4 +17,18 @@ public static AndCondition of(final QueryCondition... conditions) {
 		return new AndCondition(List.of(conditions));
 	}
 
+	@Override
+	public FilterFunction<Row> getCondition() {
+		return (Row row) -> this.conditions.stream()
+				.map(QueryCondition::getCondition)
+				.map((filterFunction) -> {
+					try {
+						return filterFunction.call(row);
+					} catch (Exception e) {
+						e.printStackTrace();
+						return false;
+					}
+				})
+				.reduce(true, (a, b) -> a && b);
```

Comment:
Nice but there is a small performance flaw here. `#reduce` does not have any knowledge of what you are trying to do. As a consequence, it will test every condition, even when one of them is true.
To short-circuit the execution, like standard AND operators do, you should use `Stream#allMatch` as follow:
```suggestion
				.allMatch((filterFunction) -> {
					try {
						return filterFunction.call(row);
					} catch (Exception e) {
						e.printStackTrace();
						return false;
					}
				});
```